crates/graphql/src/schema/objects/marketplace.rs

```diff
@@ -103,4 +105,80 @@ impl Marketplace {
             .await
             .map_err(Into::into)
     }
+
+    #[graphql(arguments(
+        wallet(description = "wallet of current user"),
+        creators(description = "creators of nfts"),
+        listed(description = "addresses of auction house")
+    ))]
+    fn nft_counts(
+        &self,
+        wallet: PublicKey<Wallet>,
+        creators: Vec<PublicKey<Wallet>>,
+        listed: Vec<PublicKey<Wallet>>,
+    ) -> NftCounts {
+        NftCounts::new(wallet, creators, listed)
+    }
+}
+
+#[derive(Debug, Clone)]
+struct NftCounts {
+    wallet: PublicKey<Wallet>,
+    creators: Vec<PublicKey<Wallet>>,
+    listed: Vec<PublicKey<Wallet>>,
+}
+
+impl NftCounts {
+    #[must_use]
+    pub fn new(
+        wallet: PublicKey<Wallet>,
+        creators: Vec<PublicKey<Wallet>>,
+        listed: Vec<PublicKey<Wallet>>,
+    ) -> Self {
+        Self {
+            wallet,
+            creators,
+            listed,
+        }
+    }
+}
+
+#[graphql_object(Context = AppContext)]
+impl NftCounts {
+    fn total(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators = &self
+            .creators
+            .map(|a| a.into_iter().map(Into::into).collect());
+        let count = nft_count::total(&conn, creators)?;
+        Ok(count.try_into()?)
+    }
+
+    fn listed(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators = &self
+            .creators
+            .map(|a| a.into_iter().map(Into::into).collect());
```

Comment:
```suggestion
        let creators = &self
            .creators
            .into_iter()
            .map(|a| a.to_string())
            .collect::<Vec<_>>();
```

transform the vec into a collection then map over and re-gather the results.